manifests/init.pp

```diff
@@ -47,31 +47,30 @@
   $authz_regexp            = {},
   $conf_file               = $openldap_slapd::params::conf_file,
   $databases               = {},
-  $global_acls             = $openldap_slapd::params::global_acls,
-  $local_ssf               = $openldap_slapd::params::local_ssf,
+  $global_acls             = undef,
+  $local_ssf               = undef,
   $log_level               = $openldap_slapd::params::log_level,
   $modules                 = [],
-  $password_hash           = $openldap_slapd::params::password_hash,
-  $password_salt_format    = $openldap_slapd::params::password_salt_format,
+  $password_hash           = undef,
```

Comment:
We cannot undef some of these defaults, as some of them aren't surrounded by conditionals in the template...  eg `password_hash`..

https://github.com/crayfishx/puppet-openldap_slapd/blob/master/templates/_security.erb#L4

This will cause the template to spit out just 

```
password-hash
```

or 

```
password-hash :undef
```

Depending on which version of Puppet is being used.   

The manifest changes are fine but we need to check how each one is handled in the template.  Keep in mind the behaviour differences between puppet 3 and 4 when it comes to handling "undef" variables in ERB templates....